### PR TITLE
Add workflow for pushing to IBM cloud registry

### PR DESCRIPTION
**Pull request by delapuente**
The new GitHub action `preview` builds the Dockerfile at the root of the repository, pushes it to IBM Container Registry, and uses it to create an application in Code Engine.

The `Dockerfile` is written so it can leverage the registry cache, copying just the dependencies of the following `RUN` command. When the `RUN` command depends on more than one file or dir, these are copied keeping the most likely to change at the end.

---

The current version is not ready to be merged yet. Before merging, we need to address:
- [x] There exists a workflow that, upon closing the PR, removes the image from the IBM Container Registry, and deletes the Code Engine application with the preview.
- [x] Review the [last step (`uses: chrnorm/deployment-action@releases/v1`)](https://github.com/marketplace/actions/deployment-action) in charge of creating a deployment _a-la_ Heroku. It seems it creates it but it is not showing up in the  PR.
- [x] Don't be lazy like me and use: https://github.com/nektos/act for recreating actions locally if it is of some help.

---

Related:
- https://github.com/qiskit-community/platypus/pull/134
- https://github.com/Qiskit/qiskit.org/issues/2149
- https://github.com/Qiskit/qiskit.org/issues/2134

### CODE REVIEW

**Comment by abdonrd** [comment]
If can helps, in the Nuxt documentation there is a few example about how to containerize the application.

[Example with Google Cloud Run](https://nuxtjs.org/docs/2.x/deployment/deployment-cloud-run#containerize-your-application):

```
FROM node:14

WORKDIR /usr/src/app

COPY . ./
RUN npm install

EXPOSE 8080

ENV HOST=0.0.0.0
ENV PORT=8080

RUN npm run build

CMD [ "npm", "run", "start" ]
```

**Comment by vabarbosa** [comment]
Deploying PR for preview

**Comment by vabarbosa** [comment]
Deploying PR for preview

**Comment by vabarbosa** [comment]
Preview available at https://qiskit-org-pr-2127.dcq4xc5i083.us-south.codeengine.appdomain.cloud

**Comment by vabarbosa** [comment]
Attempting to deploy PR for preview

**Comment by vabarbosa** [comment]
Attempting to deploy PR for preview

**Comment by vabarbosa** [comment]
Deployment successful!
Preview available at https://qiskit-org-pr-2127.dcq4xc5i083.us-south.codeengine.appdomain.cloud

**Comment by vabarbosa** [comment]
closing PR to test github action (will re-open)

**Comment by vabarbosa** [comment]
closing PR to test github action (will re-open)

**Comment by vabarbosa** [comment]
closing PR to test github action (will re-open)

**Comment by vabarbosa** [comment]
closing PR to test github action (will re-open)

**Comment by vabarbosa** [comment]
closing PR to test github action (will re-open)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -174,11 +174,12 @@ branch. In these occassions, remember to manually close the related pull request
...
-As part of our continuous integration infrastructure, every pull request receives a
-dedicated URL by [Vercel](https://vercel.com/) to preview the changes in it.
+As part of our continuous integration infrastructure, every pull request that passes 
+the build process, receives a dedicated deployment running on [IBM Code Engine](https://cloud.ibm.com/codeengine/overview). 
 
-When the contribution comes from an external contributor, previews require authorization
-from one of the core members of the team.
+This allows the team to have live, branch previews, making it easier for the team to 
+share links and review changes as necessary. You can preview your working branch at 
+`https://qiskit-org-pr-<pull-request-number>.<unique_id>.us-south.codeengine.appdomain.cloud/`.
```
Comment (eddybrando) [inline CONTRIBUTING.md:182]:
```suggestion
This allows the team to have live branch previews, making it easier to share
links and review changes as necessary. You can preview your working branch at 
`https://qiskit-org-pr-<pull-request-number>.<unique_id>.us-south.codeengine.appdomain.cloud/`.
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+     nuxt.config.ts stylelint.config.js tsconfig.json \
+     ./
+COPY tests tests/
+COPY components components/
+RUN npm run build
+
+EXPOSE 3000
+ENV HOST=0.0.0.0
+ENV NODE_ENV=development
+CMD [ "npx", "nuxt", "start" ]
```
Comment (eddybrando) [inline Dockerfile:31]: Adding newline at the end of file. (https://unix.stackexchange.com/a/18789)
```suggestion
CMD [ "npx", "nuxt", "start" ]

```

**Review by eddybrando — APPROVED** [review]
Thanks a lot for this PR!

I have only **one additional question**:

Is there a better way to copy the assets in the **Dockerfile** than selecting them one by one?

This seems like it could cause trouble for future maintenance.

Could we use `COPY .`, or maybe add the files and directories that we want to exclude to a `.dockerignore` file?

**Review by abdonrd — APPROVED** [review]
Looks really great, thanks @vabarbosa! 👏🏻 

Would it make sense to merge the two jobs (`preview` & `cleanup`) into a single worflow and use the `if` property to choose when to run it? For trying not to repeat so much in both workflows.

**Comment by vabarbosa** [comment]
> Thanks a lot for this PR!
> 
> I have only **one additional question**:
> 
> Is there a better way to copy the assets in the **Dockerfile** than selecting them one by one?
> 
> This seems like it could cause trouble for future maintenance.
> 
> Could we use `COPY .`, or maybe add the files and directories that we want to exclude to a `.dockerignore` file?

@eddybrando thanks for the comment. the `Dockerfile` is written so it can leverage the registry cache (i.e., using more specific `COPY` to limit cache busts), keeping the most likely files to change at the end.

**Comment by vabarbosa** [comment]
> Would it make sense to merge the two jobs (`preview` & `cleanup`) into a single worflow and use the `if` property to choose when to run it? For trying not to repeat so much in both workflows.

@abdonrd thanks, yeah we can definitely combine the two jobs into a single workflow. i initially thought perhaps the workflow file may become too long/unwieldly.

**Comment by delapuente** [comment]
A trick by @ismaelfaro speeding up the starting time: right after deploying, make a `curl` with a long timeout to force the "pod" to start.

**Pull request merged**